hs_access_control/models/user.py

```diff
@@ -968,41 +1061,63 @@ def edit_resources(self):
         if not self.user.is_active:
             raise PermissionDenied("Requesting user is not active")
 
-        return BaseResource.objects.filter(Q(raccess__immutable=False) &
-                                           (Q(r2urp__user=self.user,
-                                              r2urp__privilege__lte=PrivilegeCodes.CHANGE) |
-                                            Q(r2grp__group__gaccess__active=True,
-                                                r2grp__group__g2ugp__user=self.user,
-                                              r2grp__privilege__lte=PrivilegeCodes.CHANGE)))\
-                                   .distinct()
+        # This is a mouthful.
+        # a resource is editable if
+        # 1. it's shared with the user and editable.
+        # 2. it's shared with a group that has edit privilege and contains the user,
+        # 3. it's shared with a group that has edit privilege and a community that
+        #    contains the user, and the community share preserves edit
+
+        return BaseResource.objects.filter(
```

Comment:
May be if we implement a custom QuerySet with custom APIs for the BaseResource class, we can then do:
`return BaseResource.objects.mutables().user_resources(self.user, PrivilegeCodes.CHANGE).distinct() | BaseResource.objects.mutables().group_user_resources(self.user, PrivilegeCodes.CHANGE).distinct() | BaseResource.objects.mutables().community_user_resources(self.user, PrivilegeCodes.CHANGE).distinct()`

If this works, it would increase code readability considerably.

Reply:
yes and no. Abstraction makes things readable at the expense of saying what's happening :)